src/core/types/ui_settings.ts

```diff
@@ -80,6 +81,13 @@ export interface UiSettingsParams<T = unknown> {
    * Used to validate value on write and read.
    */
   schema: Type<T>;
+  /**
+   * Metric to track once this property changes
+   */
+  metric?: {
```

Comment:
@majagrubic could you mark this property as `@deprecated` and add a comment:
`a temporary measurement until https://github.com/elastic/kibana/issues/83084`